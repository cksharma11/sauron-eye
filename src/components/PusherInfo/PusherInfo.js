import React from "react";
import { Link } from "react-router-dom";

const PusherInfo = ({ pusher, number }) => {
  const { name, sha, passed, total, project, latestPushTime } = pusher;
  return (
    <tr className={`${number % 2 === 0 ? "white" : "whitesomke"}`}>
      <td className="serial-number">{number}</td>
      <td>
        <Link to={`history?username=${name}`}>{name}</Link>
      </td>
      <td>
        <a
          href={`https://github.com/step-batch-7/${project}-${name}/commit/${sha}`}
        >
          {sha.substr(0, 6)}
        </a>
      </td>
      <td>{`${passed}/${total}`}</td>
      <td>{latestPushTime}</td>
    </tr>
  );
};

export default PusherInfo;
