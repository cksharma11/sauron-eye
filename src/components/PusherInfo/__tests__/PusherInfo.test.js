import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import PusherInfo from "../PusherInfo";

configure({ adapter: new Adapter() });

describe("PusherInfo", () => {
  it("should render properly", () => {
    const pusher = {
      name: "somename",
      sha: "123sha",
      passed: 10,
      total: 15,
      project: "test"
    };
    const wrapper = shallow(<PusherInfo pusher={pusher} number={1} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
