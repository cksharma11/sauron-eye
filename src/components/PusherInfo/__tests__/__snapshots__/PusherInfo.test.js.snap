// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PusherInfo should render properly 1`] = `
"<tr className=\\"whitesomke\\">
  <td className=\\"serial-number\\">
    1
  </td>
  <td>
    <Link to=\\"history?username=somename\\">
      somename
    </Link>
  </td>
  <td>
    <a href=\\"https://github.com/step-batch-7/test-somename/commit/123sha\\">
      123sha
    </a>
  </td>
  <td>
    10/15
  </td>
  <td>
    --
  </td>
</tr>"
`;
